x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -57,40 +61,58 @@ interface State {
   savedQuery?: SavedQuery;
 }
 
+interface LensAppProps {
+  editorFrame: EditorFrameInstance;
+  data: DataPublicPluginStart;
+  navigation: NavigationPublicPluginStart;
+  core: AppMountContext['core'];
+  storage: IStorageWrapper;
+  savedObjectId?: string;
+  docStorage: SavedObjectStore;
+  redirectTo: (
+    savedObjectId?: string,
+    documentByValue?: Document,
+    returnToOrigin?: boolean,
+    newlyCreated?: boolean
+  ) => void;
+  embeddableEditorIncomingState?: EmbeddableEditorState;
+  onAppLeave: AppMountParameters['onAppLeave'];
+  history: History;
+  featureFlagConfig: FeatureFlagConfig;
+}
+
 export function App({
   editorFrame,
   data,
   core,
   storage,
-  docId,
+  savedObjectId,
   docStorage,
   redirectTo,
-  originatingApp,
+  embeddableEditorIncomingState,
   navigation,
   onAppLeave,
   history,
-}: {
-  editorFrame: EditorFrameInstance;
-  data: DataPublicPluginStart;
-  navigation: NavigationPublicPluginStart;
-  core: AppMountContext['core'];
-  storage: IStorageWrapper;
-  docId?: string;
-  docStorage: SavedObjectStore;
-  redirectTo: (id?: string, returnToOrigin?: boolean, newlyCreated?: boolean) => void;
-  originatingApp?: string | undefined;
-  onAppLeave: AppMountParameters['onAppLeave'];
-  history: History;
-}) {
+  featureFlagConfig,
+}: LensAppProps) {
   const language =
     storage.get('kibana.userQueryLanguage') ||
     core.uiSettings.get(UI_SETTINGS.SEARCH_QUERY_LANGUAGE);
 
+  const editFromDashMode =
+    !!embeddableEditorIncomingState?.originatingApp &&
```

Comment:
would it not be enough to check if `originatApp` is dashboards and then we can get rid of the second condition?

Reply:
I actually renamed this to `editFromContainerMode`, because in theory this should work regardless of the container. It's supposed to represent that we've come from another app, and are editing an embeddable (as opposed to creating one)